frontend/test/pytest/test_measurement_primitives.py

```diff
@@ -25,19 +26,172 @@
     state_p,
     var_p,
 )
+from catalyst.jax_tracer import lower_jaxpr_to_mlir
+import pytest
 
-
+@pytest.mark.xfail(reason="[WIP] Convert to lit test")
 def test_sample():
     """Test that the sample primitive can be captured into jaxpr."""
 
     def f():
         obs = compbasis_p.bind()
-        return sample_p.bind(obs, shots=5, shape=(5, 0))
+        return sample_p.bind(obs, 5, shape=(5, 0))
 
     jaxpr = jax.make_jaxpr(f)().jaxpr
-    assert jaxpr.eqns[1].primitive == sample_p
-    assert jaxpr.eqns[1].params == {"shape": (5, 0), "shots": 5}
-    assert jaxpr.eqns[1].outvars[0].aval.shape == (5, 0)
+    mlir = lower_jaxpr_to_mlir(jax.make_jaxpr(f)(), "foo")[0]
+    assert (
+        jaxpr
+        == """
+{ lambda ; . let
+    a:AbstractObs(num_qubits=0,primitive=compbasis) = compbasis
+    b:f64[5,0] = sample[shape=(5, 0)] a 5
+  in (b,) }
+"""
+    )
+
+    assert (
+        mlir
+        == """
+module @foo {
```

Comment:
I think in this file it would be nice to check specific lines and not a full function